### ISSUE TITLE
Naming conventions

### ISSUE THREAD

**Issue by GCortijo1** [dotnet/docs#15921]
I was reading your previous documentation about naming conventions and I found that now you are following a different convention.
For example, in your documentation: 
"Do capitalize both characters of two-character acronyms, except the first word of a camel-cased identifier.

A property named DBRate is an example of a short acronym (DB) used as the first word of a Pascal-cased identifier. A parameter named ioChannel is an example of a short acronym (IO) used as the first word of a camel-cased identifier."

So yeah, checking the system classes we can see is working like that, for example "DBNull", but looks like in the new ones you are following a different approach capitalizing only the first letter of the acronyms like in "DbContext".
Which is the right way? Could you update the documentation?

But

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d49bc05d-7c8e-5f89-aac5-d133225da4d5
* Version Independent ID: 2f831fc9-dfe6-15db-4ec6-00f59c2dacc2
* Content: [.NET Core Guide](https://docs.microsoft.com/en-us/dotnet/core/#feedback)
* Content Source: [docs/core/index.md](https://github.com/dotnet/docs/blob/master/docs/core/index.md)
* Product: **dotnet-core**
* GitHub Login: @richlander
* Microsoft Alias: **mairaw**

**Comment by Youssef1313** [comment]
I think it's more about the entity framework itself more than it's a doc issue (also it's not related at all to the .NET Core guide article. but I think you're correct, there is an inconsistency using `DB` like in `DBNull` and `Db` like in `DbContext`.
/cc ajcvickers

**Comment by nminaya** [comment]
It would also be weird "fix" this inconsistency with an empty class DBContext that inherits from DbContext 😅

**Comment by mairaw** [comment]
Tagging a few folks to see if they comment on the current naming conventions
@terrajobst @KrzysztofCwalina @bartonjs

**Comment by terrajobst** [comment]
@bartonjs probably knows it best, but I think the guidance is correct but there are simply outliers like the EF example. FWIW, this guidelines was never great as it was too complicated.

**Comment by mairaw** [comment]
Thanks @terrajobst. I'll close this for now but if @bartonjs has anything to add, feel free. I think this answers it.

**Comment by bartonjs** [comment]
I agree that the EF type is named incorrectly, given our guidance and the precedence of `DBNull` (and possibly others).  It looks like "Db" is the only Upper-Lower pair that is an outlier in type names.

* "Id" is special cased, even though "id" (as in `!ego && !superego`) is a different word.
* "Um" and "Al" are part of a proper noun.
* "Co" is part of C++ "coclass", which PascalCases to "CoClass", and not an acronym

---

* If "DB" is an acronym, it should be "DB".
* If "DB" is a non-acronym abbreviation, it should be "Database".
* If "Db" is a unit (which it isn't, since decibel (0.1 Bel) is dB, so decabel (10 Bel) would be DB) it should be expanded to avoid debate (and confusion).
* I can't come up with any legitimate use of "Db".

---

```
Al: class UmAlQuraCalendar
As: class MarshalAsAttribute
As: class RecommendedAsConfigurableAttribute
As: class SecurityTreatAsSafeAttribute
Be: class WaitHandleCannotBeOpenedException
By: class DivideByZeroException
By: class IsByRefLikeAttribute
By: class IsByValue
By: class MarshalByRefObject
By: class MarshalByValueComponent
By: enum GroupByBehavior
Co: class CoClassAttribute
Db: class DbColumn
Db: class DbCommand
Db: class DbCommandBuilder
Db: class DbConnection
Db: class DbConnectionStringBuilder
Db: class DbDataAdapter
Db: class DbDataReader
Db: class DbDataReaderExtensions
Db: class DbDataRecord
Db: class DbDataSourceEnumerator
Db: class DbEnumerator
Db: class DbException
Db: class DbMetaDataCollectionNames
Db: class DbMetaDataColumnNames
Db: class DbParameter
Db: class DbParameterCollection
Db: class DbProviderFactories
Db: class DbProviderFactory
Db: class DbProviderSpecificTypePropertyAttribute
Db: class DbTransaction
Db: enum DbType
Db: enum SqlDbType
Do: class DoWorkEventArgs
Id: class ApplicationId
Id: class AssemblyAlgorithmIdAttribute
Id: class DispIdAttribute
Id: class ProgIdAttribute
Id: class UniqueId
If: class DoesNotReturnIfAttribute
If: class NotNullIfNotNullAttribute
In: class InAttribute
In: class InRowChangingEventException
In: class RowNotInTableException
In: class TransactionInDoubtException
Is: class CriticalHandleMinusOneIsInvalid
Is: class CriticalHandleZeroOrMinusOneIsInvalid
Is: class IsBoxed
Is: class IsConst
Is: class IsCopyConstructed
Is: class IsExplicitlyDereferenced
Is: class IsImplicitlyDereferenced
Is: class IsJitIntrinsic
Is: class IsLong
Is: class IsPinned
Is: class IsReadOnlyAttribute
Is: class IsSignUnspecifiedByte
Is: class IsUdtReturn
Is: class IsVolatile
Is: class SafeHandleMinusOneIsInvalid
Is: class SafeHandleZeroOrMinusOneIsInvalid
No: class NoNullAllowedException
Of: class ArgumentOutOfRangeException
Of: class EndOfStreamException
Of: class IndexOutOfRangeException
Of: class OutOfMemoryException
Of: enum DayOfWeek
On: class OnDeserializedAttribute
On: class OnDeserializingAttribute
On: class OnSerializedAttribute
On: class OnSerializingAttribute
On: enum ValueTaskSourceOnCompletedFlags
Op: class OpCodes
Op: enum OpCodeType
Op: struct OpCode
To: class InternalsVisibleToAttribute
To: class ManagedToNativeComInteropStubAttribute
To: class ToBase64Transform
To: class TypeForwardedToAttribute
Um: class UmAlQuraCalendar
```

The data was based on scrubbing from ref.cs files in dotnet/standard, then some extraction/prettification in Vim.

```
:~/git/dotnet/standard/src $ find . -wholename \*/ref/\*.cs -exec grep "\(class\|struct\|enum\) [^ :]*[A-Z][a-z][A-Z]" {} \;
    public abstract partial class CriticalHandleMinusOneIsInvalid : System.Runtime.InteropServices.CriticalHandle
    public abstract partial class CriticalHandleZeroOrMinusOneIsInvalid : System.Runtime.InteropServices.CriticalHandle
    public abstract partial class SafeHandleMinusOneIsInvalid : System.Runtime.InteropServices.SafeHandle
    public abstract partial class SafeHandleZeroOrMinusOneIsInvalid : System.Runtime.InteropServices.SafeHandle
    public enum ValueTaskSourceOnCompletedFlags
    public sealed partial class AssemblyAlgorithmIdAttribute : System.Attribute
    public partial class ToBase64Transform : System.IDisposable, System.Security.Cryptography.ICryptoTransform
    public partial class DoWorkEventArgs : System.ComponentModel.CancelEventArgs
    public partial class MarshalByValueComponent : System.ComponentModel.IComponent, System.IDisposable, System.IServiceProvider
    public partial class RecommendedAsConfigurableAttribute : System.Attribute
[...]
```

```
~/git/dotnet/standard/src $ find . -wholename \*/ref/\*.cs -exec grep "\(class\|struct\|enum\) [^ :]*[A-Z][a-z]\>" {} \;
    public partial class UniqueId
    public sealed partial class ApplicationId
```

**Comment by ajcvickers** [comment]
We looked at the existing non-EF classes in .NET when we named `DbContext` back in 2010. Pretty much everything from ADO.NET (pre-EF) such as DbConnection, DbCommand, and so on used the "Db" casing. Only `DBNull` was the outlier. So, regardless of what the _de jury_ naming convention said, after days and days of discussion it was decided to continue with the _de facto_ convention of using "Db".

Why "Db" was originally chosen for ADO.NET is unknown to me--before my time. However, I have to say I always thought the exception to the normal rule for only two letters was completely unnecessary, and `DBContext` would have been hurting my eyes for the last ten years or so.

**Comment by terrajobst** [comment]
Sure, but we should not repeat this moving forward. Within EF, let's stick to `Db`. But let's not add new two-letter exceptions :-)

**Comment by ajcvickers** [comment]
@terrajobst To be clear, most of the `Db` types are not EF types. They are System.Data.Common types in the BCL. I think we should stick to `Db` in those types as well.

**Comment by KrzysztofCwalina** [comment]
> I always thought the exception to the normal rule for only two letters was completely unnecessary,

The two letter acronym rule was added so people don't think that the System.IO namespace is referring to a moon.

**Comment by ajcvickers** [comment]
@KrzysztofCwalina Well it's a huge, huge relief that we avoided that massive source of potential confusion!
